_docs/0.3/usage/dataset-comparison.md

```diff
@@ -0,0 +1,135 @@
+
+##### Default properties are
+
+```json
+dataset-comparison {
+  errColumn = "errCol"
+  actualPrefix = "actual"
+  expectedPrefix = "expected"
+  allowDuplicates = false
+}
+```
+
+#### Parameters
+
+Dataset Comparison can recieve two types of parameters + `keys` and `out-path`. There are input general, input specific, `keys` and `out-path`. All should be prefixed with `--` as is standard in command line tools to denote parameter/key followed by its value.
+
+- General parameters are those that will be applied to both reference and new data. These have no prefix
+- Specific parameters are those that differentiate if you want to use them for reference or new data source. They are prefixed eaither with `ref-` or `new-`. They override the general ones
+- `keys` is a list of strings delimetered by a `,` (comma) that represent the primary key(s)
+- `out-path` is a path to a directory, that will be created by the job, and will hold the diff dataframe and `_METRICS` file
+
+Input parameters (be it general or specific) are governed by the spark datasource formats and options. Imagine having a scala spark code:
+
+```scala
+sparkSession
+    .read
+    .format(<format specified>)
+    .options(<map of options provided>)
+    .load(<path provided>)
+```
+
+Same way you would provide the Dataset Comparison spark job with `format`, but prefix it with `--`, `path`, but prefix it if either `--ref-` or `--new-`, dependeing on the origing. Maybe add a `--header` option, if you chose `CSV` format or `--rowTag`, if you chose `XML`.
+
+Now how to use it all together:
+
+- `--format`, `--new-format`, `--ref-format` for choosing the format type. This can be `parquet`, `csv`, `xml`, `jdbc`, etc.
+- `--ref-path`, `--new-path` and `--out-path` define paths on HDFS of reference data path, newly created data path and diff output path, repspectively. In case of usage of `jdbc` format these can be replaced by `--ref-dbtable` and `--ned-dbtable`
+- `--keys` to specifie the primary key(s) of the table, e.g. `--keys ID1, ID2`. These can be only top level columns.
+- format specific keys prefixed with `--`, so `--header`, `--rowTag`, `--user`, etc.
+
+**Mandatory fields** are logically the paths/dbtables, format (at least the general one) and any format specific keys that are prescribed by the format used.
+
+**NOTE** if you want to use `jdbc` format or others, requiring specific drivers, you should have appropriate drivers on a class path.
+
+
+#### Usage as library
+
+For usage as library use class `za.co.absa.hermes.datasetComparison.DatasetComparator`. 
+
+**Prescription of DatasetComparator class**:
+
+```scala
+/**
+ * Class that is the brain of the DatasetComparison module. This class should be used in case of using DatasetComparison
+ * as a library. In case of running the DatasetComparison as SparkJob, please use the DatasetComparisonJob.
+ *
+ * @param dataFrameReference is a Dataframe used in a comparison as the origin of truth
```

Comment:
```suggestion
 * @param dataFrameReference is the Dataframe used in a comparison as the origin of truth
```

Reply:
I would like to refuse this because this is one to one copy with the code. Will try and keep in mind

Reply:
How is this (and the other similar) a one-to-one copy with the code? I thought it was a grammatical thing. But happy to ask a native speaker to judge this.